app/models/mixins/service_mixin.rb

```diff
@@ -42,20 +41,15 @@ def remove_all_resources
   end
 
   def max_group_delay(grp_idx, delay_type)
-    result = 0
-    each_group_resource(grp_idx) { |r| result = [result, r[delay_type] || self.class::DEFAULT_PROCESS_DELAY_BETWEEN_GROUPS].max }
-    result
+    each_group_resource(grp_idx).collect { |r| r[delay_type] || self.class::DEFAULT_PROCESS_DELAY_BETWEEN_GROUPS }.max.to_i
   end
 
   def delay_for_action(grp_idx, action)
     max_group_delay(grp_idx, delay_type(action))
   end
 
   def combined_group_delay(action)
-    group_idxs = service_resources.map(&:group_idx).uniq
-    [].tap do |results|
-      group_idxs.each { |idx| results << max_group_delay(idx, delay_type(action)) }
-    end.sum
+    service_resources.map(&:group_idx).uniq.collect { |idx| max_group_delay(idx, delay_type(action)) }.sum
```

Comment:
Small nit, but could probably just change the `collect` to `map` or change them all to `collect`.